src/eme.js

```diff
@@ -116,12 +116,7 @@ export const makeNewRequest = (player, requestOptions) => {
     });
 
     player.on('dispose', () => {
-      keySession.close().then(() => {
-        eventBus.trigger({
-          type: 'keysessionclosed',
-          keySession
-        });
-      }).catch((error) => {
```

Comment:
Since we're doing this on `dispose`, removing the `then()` makes the most sense. Since the `keySession.close()` is async, we will have disposed the tech (i.e. `eventBus`) by the time this promise resolves.

Future enhancement might be to delay the `dispose` until the key session is closed, but I'm not sure that would be useful or necessary.

Reply:
Do we want to trigger the `keysessionclosed` event anyway outside of the `then` block?

Reply:
I am not sure about this change, maybe it will be safer to just: replace all occurrences of the `eventBus.trigger(...args)` to the `trigger(eventBus, ...args)`

Where `trigger` is

```js

function trigger(eventBus, ...args) {
  if (eventBus.isDisposed()) {
     return;
  }

  eventBus.trigger(...args);
}

```